### DISCUSSION TITLE
Grouping dependencies in the same directory with additionalBranchPrefix already configured

### DISCUSSION THREAD

**Discussion by zephyros-dev** [renovatebot/renovate#21336] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have a monorepo that look like this:
```
.
├── development/
│   ├── service-1/
│   │   ├── dep-1
│   │   └── dep-2
│   └── service-2/
│       ├── dep-1
│       └── dep-3
└── production/
    ├── service-1/
    │   ├── dep-1
    │   └── dep-2
    └── service-2/
        ├── dep-1
        └── dep-3
```
I have configured the renovate so that it can make the same dependencies (dep-1) to have separate PR based on the environment (development,production). The configuration look something like this:
```
{
  "packageRules": [
    {
      "additionalBranchPrefix": "{{{ replace '(development|production)/.*' '$1' packageFileDir }}}/"
    }
  ]
}
```
With this config the branches are created like this:
```
/renovate/development/dep-1
/renovate/production/dep-1
```
So far so good, but what I'd like to have in addition to this is to group the updates in the same folder together, so the PR will look something like this:
```
# dep-1 get an update, 2 branches for each environment will be created
/renovate/development/dep-1
/renovate/production/dep-1
# when dep-2 get an update while dep-1 is still a PR, dep-2 is grouped with dep-1 based on the parentDir
/renovate/development/service-1
/renovate/production/service-1
```
How should I go around doing this?

### What you tried so far.

 I suppose doing group name based on the matchPaths for both `service-1` and `service-2` would work, something like this:
```
    {
      "groupName": "service-1",
      "matchPaths": ["**/service-1/**"]
    },
    {
      "groupName": "service-2",
      "matchPaths": ["**/service-2/**"]
    }
```
Unfortunately it is not scalable if you have 10 or more services, so I'm not sure how to go about doing this

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
I don't understand what you mean by the "look something like this" example. Could you try rephrasing or elaborating?

**Reply by zephyros-dev** [reply]
I rethink about the problem and realize that I must either group the dependencies updates either by folder or the dependency, since the `service-1` and `service-2` have overlap of `dep-1`, so my question here is invalid. Thank you for the prompt reply.
